### PR TITLE
Improvements and new hyperparameters

### PR DESCRIPTION
**Pull request by albanPi**
# Improvements : 
- Added a `skip` function to avoid datasets with samples over 5000 for `SVC`.
- Set `patience` to 5 instead of 3 (default) to allow more iteration (with also the contribution of `get_next` new method) and to help decision trees.
- Added a `extra_model_params` mostly for `HistGradientBoostingClassifier`'s computing issues

# Hyperparameters changes : 
New hyperparameters for some solvers that are relevant for the benchmark. Also adjustment of hyerparameters for `SVC` and `LogisticRegression`

### CODE REVIEW

File: benchmark_utils/optuna_solver.py [hunk 1]
```diff
@@ -20,14 +20,16 @@
...
-        strategy='callback'
+        strategy='callback', patience=5
     )
 
     params = {
         'test_size': 0.25,
         'seed': 42,
     }
 
+    extra_model_params = {}
```
Comment (tomMoral) [inline benchmark_utils/optuna_solver.py:31]: Not sure why it is necessary as this is not used anywhere. If it is, can you add a comment?

File: benchmark_utils/optuna_solver.py [hunk 2]
```diff
@@ -95,5 +97,9 @@ def get_result(self):
         # it is customizable for each benchmark.
         return self.clf
 
+    @staticmethod
+    def get_next(n_iter):
```
Comment (tomMoral) [inline benchmark_utils/optuna_solver.py]: You don't need to have a `staticmethod` anymore with the newest version of `benchopt`
```suggestion
    def get_next(self, n_iter):

```

**Review by tomMoral — COMMENTED** [review]
This seems nice.

File: benchmark_utils/optuna_solver.py (hunk 1 above)
Comment (albanPi) [reply benchmark_utils/optuna_solver.py:31]: I don't remember why this variable was added. Hovewer, `HistGradientBoostingClassifier` has the same variable with a parameter for the preprocessor. So it's maybe for the others solvers ?

File: benchmark_utils/average_clf.py [hunk 3]
```diff
@@ -21,7 +21,8 @@ def predict(self, X_test):
         y_pred = []
         for e in self.estimators:
             y_pred.append(e.predict(X_test))
-        y_pred = np.array(y_pred)
+        # 1* convert boolean to 0 & 1 for stats.mode
+        y_pred = 1*(np.array(y_pred))
```
Comment (tomMoral) [inline benchmark_utils/average_clf.py]:
```suggestion
        # convert boolean to 0 & 1 for stats.mode
        y_pred = np.array(y_pred).astype(int)
```

File: benchmark_utils/optuna_solver.py [hunk 4]
```diff
@@ -86,7 +88,9 @@ def run(self, callback):
             best_model = AverageClassifier(
                 study.best_trial.user_attrs['model']
             )
-        self.clf = best_model
+        self.best = study.best_params
+        self.clf = self.get_model().set_params(
+            **self.best).fit(self.X_train, self.y_train)
```
Comment (tomMoral) [inline benchmark_utils/optuna_solver.py]:
```suggestion
            **self.best
        ).fit(self.X_train, self.y_train)

```

File: benchmark_utils/optuna_solver.py (hunk 1 above)
Comment (tomMoral) [reply benchmark_utils/optuna_solver.py:31]: Ok I see, now that it is used it is clearer.

**Review by tomMoral — APPROVED** [review]
LGTM

**Pull request merged**